### PR TITLE
JP-3098 switch order of precedence between PATTTYPE and SRCTYAPT in the srctype step

### PR DESCRIPTION
**Pull request by jemorrison**
Resolves [JP-3098](https://jira.stsci.edu/browse/JP-3098)

Closes #

This PR changes the order of the precedence of the rules for setting the src_type for 'MIR_LRS-FIXEDSLIT', 'MIR_LRS-SLITLESS', 'MIR_MRS', 'NRC_TSGRISM', 'NRS_FIXEDSLIT', 'NRS_BRIGHTOBJ', 'NRS_IFU']. The SRCTYPE has precedence over the PATTTYPE. 

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by jemorrison** [comment]
Running regression tests - run 717. Will post results when finished

File: CHANGES.rst [hunk 1]
```diff
@@ -69,6 +69,12 @@ ramp_fitting
   of groups in a segment are computed when applying optimal weighting to line
   fit segments. [#7560, spacetelescope/stcal#163]
 
+srctype
+-------
+
+- The SRCTYPE takes precedence over PATTTYPE when setting the source type for
```
Comment (hbushouse) [inline CHANGES.rst:75]: I believe you mean SRCTYAPT here, as opposed to SRCTYPE? The former is the input to the step, the latter is the output.

**Review by hbushouse — COMMENTED** [review]
Looks good overall. Just one minor comment.

**Comment by jemorrison** [comment]
results of regression test show no differences. 
https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/717/

This seems reasonable given usually the PATTTYPE is in line with SCRTYAPT in the regression tests.

**Review by hbushouse — APPROVED** [review]

**Pull request merged**